test/testing_utils.dart

```diff
@@ -0,0 +1,224 @@
+import 'angular_utils.dart';
+import 'dart:math';
+import 'package:webdriver/io.dart';
+import 'package:random_string/random_string.dart' as random;
+import 'dart:async';
+import 'dart:io';
+
+void startSelenium() {
+  print("Starting Selenium Chrome Driver");
+  Process.run(
+      'java.exe -Ddriver.chrome.driver=chromedriver.exe -jar '
+      'selenium-server-standalone-3.6.0.jar',
+      []);
+}
+
+Future<WebDriver> setupWebDriver(int width, int height) async {
+  var options = {'browserName': 'chrome'};
+  WebDriver driver = await createDriver(desired: options);
+  driver.logs
+      .get('browser')
+      .where((LogEntry l) => l.level == 'SEVERE')
+      .listen((LogEntry l) => print(l));
+  await (await driver.window).setSize(new Rectangle<int>(0, 0, width, height));
+  return driver;
+}
+
+Future navigateToPage(WebDriver driver, String slug) async {
+  await driver.get("https://www.gyerunkanyukam.hu/" + slug);
+}
+
+Future<String> signUpNewRandomUser(WebDriver driver, String pw) async {
+  var newRandomUserEmail;
+  var loginBtn = await driver
+      .findElement(const By.cssSelector('div.menu-item.menu-item--login > a'));
```

Comment:
All `Finder`s should be implemented in Page Objects. Tests should only contain test logic and have all DOM locator functionality detached in POs. A good and short read on this pattern: https://martinfowler.com/bliki/PageObject.html

Ideally page objects should represent a section of a screen, and should be shared among tests.  Here is a minimalist example of how this specific selector could be moved to a PO:

```
class POLoginScreen {
  @inject
  PageLoader loader;

  @ByCss('div.menu-item.menu-item--login a')
  PageLoaderElement loginBtn;

  Future gotoLoginPage() async {
    return angularCall(loginBtn.click);
  }
}
```

Reply:
Another note: avoid using the child combinator `>` selector whenever possible - it makes `Finders` more susceptible to break on layout changes .